pkg/pillar/zedcloud/send.go

```diff
@@ -141,19 +150,21 @@ func VerifyAllIntf(ctx ZedCloudContext,
 				// We have enough uplinks with cloud connectivity working.
 				break
 			}
-			resp, _, rtf, err := SendOnIntf(ctx, url, intf, 0, nil, allowProxy)
+			resp, _, rtf, err := SendOnIntf(*ctx, url, intf, 0, nil, allowProxy)
 			if rtf {
 				remoteTemporaryFailure = true
 			}
 			if err != nil {
 				log.Errorf("Zedcloud un-reachable via interface %s: %s",
 					intf, err)
 				errorList = append(errorList, err)
+				ctx.DeviceNetworkStatus.SetPortErrorByIfname(intf, err)
```

Comment:
If rtf then you shouldn't mark it as failed. That is an indication that we have some certificate issue at the remote server end, hence doesn't indicate an issue with the port.

Reply:
Changed VerifyAllIntf to return a map of per-interface errors as discussed.